client/go/vespa/target.go

```diff
@@ -239,38 +273,42 @@ func (t *cloudTarget) resolveEndpoint(cluster string) (string, error) {
 	return "", fmt.Errorf("no endpoints")
 }
 
-func (t *cloudTarget) Type() string { return t.targetType }
+func (t *cloudTarget) Type() string {
+	switch t.apiOptions.System.Name {
+	case MainSystem.Name, CDSystem.Name:
+		return TargetHosted
+	}
+	return TargetCloud
+}
+
+func (t *cloudTarget) Deployment() Deployment { return t.deploymentOptions.Deployment }
 
 func (t *cloudTarget) Service(name string, timeout time.Duration, runID int64, cluster string) (*Service, error) {
-	if name != deployService && t.urlsByCluster == nil {
+	if name != deployService && t.deploymentOptions.ClusterURLs == nil {
 		if err := t.waitForEndpoints(timeout, runID); err != nil {
 			return nil, err
 		}
 	}
 	switch name {
 	case deployService:
-		return &Service{Name: name, BaseURL: t.apiURL}, nil
-	case queryService:
-		queryURL, err := t.resolveEndpoint(cluster)
-		if err != nil {
-			return nil, err
-		}
-		return &Service{Name: name, BaseURL: queryURL, TLSOptions: t.tlsOptions}, nil
-	case documentService:
-		documentURL, err := t.resolveEndpoint(cluster)
+		return &Service{Name: name, BaseURL: t.apiOptions.System.URL, TLSOptions: t.apiOptions.TLSOptions, ztsClient: t.ztsClient}, nil
+	case queryService, documentService:
+		url, err := t.resolveEndpoint(cluster)
 		if err != nil {
 			return nil, err
 		}
-		return &Service{Name: name, BaseURL: documentURL, TLSOptions: t.tlsOptions}, nil
+		t.deploymentOptions.TLSOptions.AthenzDomain = t.apiOptions.System.AthenzDomain
+		return &Service{Name: name, BaseURL: url, TLSOptions: t.deploymentOptions.TLSOptions, ztsClient: t.ztsClient}, nil
 	}
 	return nil, fmt.Errorf("unknown service: %s", name)
 }
 
-// SignRequest adds authentication data to a http.Request.
-// The api key is used if set on cloudTarget, if not the Auth0 device flow is used.
 func (t *cloudTarget) SignRequest(req *http.Request, sigKeyId string) error {
-	if t.apiKey != nil {
-		signer := NewRequestSigner(sigKeyId, t.apiKey)
+	if t.apiOptions.TLSOptions.KeyPair.Certificate != nil {
```

Comment:
If I understand correctly authentication in `main` and `public` is completely different.
Could we make that a bit more clean in this function? mTLS for instance won't work against public. IF nothing else some sanity checking around what work where could be good?